Diagnostics/HealthChecker/DataCollection/ExchangeInformation/Get-ExchangeServerCertificates.ps1

```diff
@@ -61,10 +61,59 @@
                         $certStatus = ($cert.Status).ToString()
                     }
 
+                    if ([String]::IsNullOrEmpty($cert.SignatureAlgorithm.FriendlyName)) {
+                        $certSignatureAlgorithm = "Unknown"
+                        $certSignatureHashAlgorithm = "Unknown"
+                        $certSignatureHashAlgorithmSecure = 0
+                    } else {
+                        $certSignatureAlgorithm = $cert.SignatureAlgorithm.FriendlyName
+                        <#
+                            OID Table
+                            https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-gpnap/a48b02b2-2a10-4eb0-bed4-1807a6d2f5ad
+                            SignatureHashAlgorithmSecure = Unknown 0
+                            SignatureHashAlgorithmSecure = Insecure/Weak 1
+                            SignatureHashAlgorithmSecure = Secure 2
+                        #>
+                        switch ($cert.SignatureAlgorithm.Value) {
```

Comment:
Is there a reason why we are doing `value` over the `FriendlyName`?

`FriendlyName` seems a little more straight forward to quickly determine if it is secure or not.

Reply:
I decided to use the Value (OID) to make this check more reliable (compared to parsing the FriendlyName). The OIDs are standardised. Yes, it's true that this means a little bit more overhead, but I think it's worth it. There are also some FriendlyNames which do not indicate the SHA (like `specifiedECDSA` which makes use of SHA256).

Reply:
I have never tried it this way. Makes sense. I'll try it out.

Reply:
As discussed - unfortunately not possible in PS.